packages/block-editor/src/components/block-pattern-setup/index.js

```diff
@@ -22,43 +20,106 @@ import BlockPreview from '../block-preview';
 import SetupToolbar from './setup-toolbar';
 import usePatternsSetup from './use-patterns-setup';
 import { VIEWMODES } from './constants';
+import { unlock } from '../../lock-unlock';
+
+const {
+	CompositeV2: Composite,
+	CompositeItemV2: CompositeItem,
+	useCompositeStoreV2: useCompositeStore,
+} = unlock( componentsPrivateApis );
 
 const SetupContent = ( {
 	viewMode,
 	activeSlide,
 	patterns,
+	onChangeSlide,
 	onBlockPatternSelect,
 	showTitles,
 } ) => {
-	const composite = useCompositeState();
 	const containerClass = 'block-editor-block-pattern-setup__container';
-	if ( viewMode === VIEWMODES.carousel ) {
-		const slideClass = new Map( [
-			[ activeSlide, 'active-slide' ],
-			[ activeSlide - 1, 'previous-slide' ],
-			[ activeSlide + 1, 'next-slide' ],
-		] );
-		return (
-			<div className="block-editor-block-pattern-setup__carousel">
-				<div className={ containerClass }>
-					<ul className="carousel-container">
-						{ patterns.map( ( pattern, index ) => (
-							<BlockPatternSlide
-								className={ slideClass.get( index ) || '' }
-								key={ pattern.name }
-								pattern={ pattern }
-							/>
-						) ) }
-					</ul>
+	const commonProps = { containerClass, patterns };
+
+	return viewMode === VIEWMODES.carousel ? (
+		<SetupContentAsCarousel
+			{ ...commonProps }
+			activeSlide={ activeSlide }
+			onChangeSlide={ onChangeSlide }
+		/>
+	) : (
+		<SetupContentAsGrid
+			{ ...commonProps }
+			onBlockPatternSelect={ onBlockPatternSelect }
+			showTitles={ showTitles }
+		/>
+	);
+};
+
+function SetupContentAsCarousel( {
+	containerClass,
+	patterns,
+	activeSlide,
+	onChangeSlide,
+} ) {
+	const activeId = `block-editor-block-pattern-setup__carousel__${ patterns[ activeSlide ].name }__tab`;
+	const composite = useCompositeStore( {
+		defaultActiveId: activeId,
+		orientation: 'horizontal',
+	} );
+	composite.setActiveId( activeId );
```

Comment:
This is necessary because `activeId` can get out of sync with `compositeStore` (by using previous/next buttons, for example), and setting it directly in `useCompositeStore` locks it completely.

Reply:
Not sure I follow entirely here — I would assume that doing something like 


```
	const composite = useCompositeStore( {
		activeId,
		orientation: 'horizontal',
	} );
```

would keep the active ID in sync with the store? Which update would cause the active ID to go out of sync with the store?

Reply:
This was related to the now-removed keyboard interaction on the carousel items. Somewhat moot now.